commit f74f94140fa50f768e61d626de4c146502b9102d
Author: Paolo Bonzini <pbonzini@redhat.com>
Date:   Thu Apr 23 13:22:27 2020 -0400

    KVM: SVM: introduce nested_run_pending
    
    We want to inject vmexits immediately from svm_check_nested_events,
    so that the interrupt/NMI window requests happen in inject_pending_event
    right after it returns.
    
    This however has the same issue as in vmx_check_nested_events, so
    introduce a nested_run_pending flag with the exact same purpose
    of delaying vmexit injection after the vmentry.
    
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

diff --git a/arch/x86/kvm/svm/svm.h b/arch/x86/kvm/svm/svm.h
index 98c2890d561d..435f3328c99c 100644
--- a/arch/x86/kvm/svm/svm.h
+++ b/arch/x86/kvm/svm/svm.h
@@ -97,6 +97,10 @@ struct nested_state {
 	/* A VMEXIT is required but not yet emulated */
 	bool exit_required;
 
+	/* A VMRUN has started but has not yet been performed, so
+	 * we cannot inject a nested vmexit yet.  */
+	bool nested_run_pending;
+
 	/* cache for intercepts of the guest */
 	u32 intercept_cr;
 	u32 intercept_dr;